src/transformStateWithClones.js

```diff
@@ -8,6 +8,36 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const actionsCopy = [];
+  const stateCopy = { ...state };
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties':
+        Object.assign(stateCopy, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          if (stateCopy.hasOwnProperty(key)) {
+            delete stateCopy[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const prop in stateCopy) {
+          delete stateCopy[prop];
+        }
+        break;
+
+      default:
```

Comment:
mind throwing some error here, so user can understand, that he gave an unknown action type

Reply:
(in `default` case)